sdk/core/core/inc/az_json_read.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_JSON_READ_H
+#define AZ_JSON_READ_H
+
+#include <az_const_str.h>
+#include <az_result.h>
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+  AZ_JSON_NO_MORE_ITEMS = AZ_JSON_RESULT + 1,
+  AZ_JSON_ERROR_UNEXPECTED_END,
+  AZ_JSON_ERROR_UNEXPECTED_CHAR,
+  AZ_JSON_ERROR_INVALID_STATE,
+  AZ_JSON_ERROR_STACK_OVERFLOW,
+};
+
+typedef az_const_str az_json_string;
+
+typedef enum {
+  AZ_JSON_NONE    = 0,
+  AZ_JSON_NULL    = 1,
+  AZ_JSON_BOOLEAN = 2,
+  AZ_JSON_NUMBER  = 3,
+  AZ_JSON_STRING  = 4,
+  AZ_JSON_OBJECT  = 5,
+  AZ_JSON_ARRAY   = 6,
+} az_json_value_tag;
+
+typedef struct {
+  az_json_value_tag tag;
+  union {
```

Comment:
this is a C11 feature.

Reply:
Do you mean anonymous unions are not allowed?

Reply:
yes. they are a C11 feature.